byway/src/main/java/com/google/sps/servlets/DestinationsServlet.java

```diff
@@ -30,52 +31,53 @@
 
   private final Gson gson = new Gson();
   private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-  private Key userKey;
-
-  @Override
-  public void init() {
-    Entity userEntity = new Entity("UserInputs");
-    userEntity.setProperty("start", "");
-    userEntity.setProperty("destinations", new ArrayList<String>());
-    datastore.put(userEntity);
-    userKey= userEntity.getKey();
-  }
-
+  private final UserService userService = UserServiceFactory.getUserService();
+  private Key tripKey;
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    tripKey = KeyFactory.stringToKey(request.getParameter("tripKey"));
+    addUserEntity(tripKey);
+
     Entity entity; 
     try{ 
-      entity = datastore.get(userKey);
+      entity = datastore.get(tripKey) ;
     } catch(EntityNotFoundException e){
-      UserLocations userLocations = new UserLocations("", new ArrayList<String>()); 
+      Trip trip = new Trip("","", new ArrayList<String>(), new ArrayList<String>(), new ArrayList<String>()); 
       response.setContentType("application/json;");
-      response.getWriter().println(gson.toJson(userLocations)); 
-      logger.atInfo().withCause(e).log("Unable to find UserLocations Entity %s", userKey);
+      response.getWriter().println(gson.toJson(trip)); 
+      logger.atInfo().withCause(e).log("Unable to find Trip Entity %s", tripKey);
       return;
     }
     String start = (String) entity.getProperty("start");
-    ArrayList<String> destinations = (ArrayList<String>) entity.getProperty("destinations");
-    UserLocations userLocations = new UserLocations(start, destinations); 
+    ArrayList<String>  destinations;
+    if (entity.getProperty("destinations")!= null){
+      destinations = (ArrayList<String>) entity.getProperty("destinations");
+    }
+    else{
+      destinations = new ArrayList<String>();
+    }
+    Trip trip = new Trip(KeyFactory.keyToString(tripKey),start, destinations, new ArrayList<String>(), new ArrayList<String>()); 
     response.setContentType("application/json;");
-    response.getWriter().println(gson.toJson(userLocations)); 
+    response.getWriter().println(gson.toJson(trip));
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException { 
     Entity entity;
+    String start = request.getParameter("start-location");
+    String destination = request.getParameter("destinations");
     try{ 
-      entity = datastore.get(userKey) ;
+      entity = datastore.get(tripKey) ;
     } catch(EntityNotFoundException e){
-      UserLocations userLocations = new UserLocations("", new ArrayList<String>()); 
+      Trip trip = new Trip("","", new ArrayList<String>(), new ArrayList<String>(), new ArrayList<String>());
```

Comment:
A post request for a nonexistent entity should probably not return a blank one. It can return an error message or nothing depending on how we want to handle misconfigured requests.